src/server/game/Grids/Notifiers/GridNotifiers.h

```diff
@@ -951,6 +951,15 @@ namespace Trinity
                 if (!u->isTargetableForAttack(false))
                     return false;
 
+                if (_exactRange)
+                {
+                    if (!i_obj->GetPosition().IsInDist(u->GetPositionX(), u->GetPositionY(), u->GetPositionZ(), i_range))
+                        return false;
+
+                    i_range = i_obj->GetPosition().GetExactDist(u->GetPosition());
+                    return true;
```

Comment:
this is skipping the checks below

Reply:
How does it skip checks if it's all in one if and block.

Reply:
why ? because "return" exits the current function scope, so the whole code below is not executed, in particular ```!i_obj->IsValidAttackTarget(u)```